### ISSUE TITLE
Instance is not woken up by Function Proxy (keeps returning 503)

### ISSUE THREAD

**Issue by scale-tone** [Azure/static-web-apps#940]
[This setup](https://github.com/scale-tone/cognitive-search-static-web-apps-sample-ui#cognitive-search-static-web-apps-sample) now suffers from Web App instance going to (some sort of) sleep and then not being properly woken up.

Specifically, when you navigate to https://lively-sand-033e9ec03.azurestaticapps.net/ after a period of inactivity, the statics are being fetched from either browser cache or CDN (and therefore do not wake up the instance), while /api requests keep failing with 503 status for quite a while (minutes):

![image](https://user-images.githubusercontent.com/5447190/193832708-345d7a12-400e-4ea5-818b-f77b1b98d96c.png)

The specifics of /api endpoint in that app is that it doesn't have any Functions, [only Proxies](https://github.com/scale-tone/cognitive-search-static-web-apps-sample-ui/blob/master/api/proxies.json). I'm speculating here, but this might be the reason why the instance is not being properly woken up.

**To Reproduce**

Go to https://lively-sand-033e9ec03.azurestaticapps.net/ in the middle of the night and press F12. Observe the Network tab.

Code is here: https://github.com/scale-tone/cognitive-search-static-web-apps-sample-ui

**Expected behavior**
Requests to /api endpoint should never fail. If the instance is asleep, they should bring it back online and return success.

**Comment by scale-tone** [comment]
ping

**Comment by thomasgauvin** [comment]
Hi @scale-tone, proxy support for Azure Functions was removed from the 4.x runtime, and as such this is no longer supported.  https://learn.microsoft.com/en-us/azure/azure-functions/functions-proxies

The recommendation is to migrate to proxying with API Management, which you can attach to your static web app by using the BYO API feature.

**Comment by scale-tone** [comment]
@thomasgauvin , 

> proxy support for Azure Functions was removed from the 4.x runtime

That's not true. [Proxies have just been re-enabled in Functions 4.x](https://learn.microsoft.com/en-us/azure/azure-functions/legacy-proxies#re-enable-proxies-in-functions-v4x), due to a [substantial demand from the community](https://github.com/Azure/Azure-Functions/issues/2047). And in fact, the above scenario is a great example when APIM wouldn't work as a replacement. Please, see the details [in the repo](https://github.com/scale-tone/cognitive-search-static-web-apps-sample-ui) [and in the issue](https://github.com/Azure/Azure-Functions/issues/2047#issuecomment-1221419316).

Besides, as I explained, in the above instance Function Proxies do work. You can try it yourself by navigating to the demo page. What doesn't work is (presumably) Static Web App's wake-up logic.

**Comment by thomasgauvin** [comment]
Thanks for bringing that up, I was unaware that support had been re-added and it is possible that our infrastructure needs to be adapted to handle this. Are you able to get it working by using a standalone Azure Function with BYO API in the meantime?

**Comment by scale-tone** [comment]
> Are you able to get it working by using a standalone Azure Function with BYO API in the meantime?

I haven't tried. 
But again, as I said, I am able to get it working even without 'Bring-Your-Own-Function'. The above setup does still works, it is just unstable.

Plus, BYOF approach would defeat the purpose of simplicity and gracefullness of the setup. Configuring, deploying and paying for yet another Azure resource just for the sake of proxying API calls through it does looks like an overkill.

**Comment by thomasgauvin** [comment]
I understand, I was hoping to provide an alternative while we investigate this issue

**Comment by Reshmi-Sriram** [comment]
Hi @scale-tone 
In order to enable proxies, you can add `SWA_ENABLE_PROXIES_MANAGED_FUNCTIONS` to `true` in your application settings,
We are working on adding this information to the documentation soon.

**Comment by scale-tone** [comment]
@Reshmi-Sriram , just to confirm that the issue is still there.

Everything is configured as you said:
![image](https://user-images.githubusercontent.com/5447190/227333762-d9cb9677-b4f4-46f2-b084-119007388793.png)

Yet still the app periodically goes offline, with proxy endpoints returning 503:
![image](https://user-images.githubusercontent.com/5447190/227334037-b66c48b9-8221-4d43-9cc6-3f83e56a18e1.png)

Any progress on this?

**Comment by Reshmi-Sriram** [comment]
> @Reshmi-Sriram , just to confirm that the issue is still there.
> 
> Everything is configured as you said: ![image](https://user-images.githubusercontent.com/5447190/227333762-d9cb9677-b4f4-46f2-b084-119007388793.png)
> 
> Yet still the app periodically goes offline, with proxy endpoints returning 503: ![image](https://user-images.githubusercontent.com/5447190/227334037-b66c48b9-8221-4d43-9cc6-3f83e56a18e1.png)
> 
> Any progress on this?

Thanks for trying it out @scale-tone!
We'd probably need to dig deeper and figure out why is it not working for you here. Adding @vivekjilla here.

**Comment by scale-tone** [comment]
@Reshmi-Sriram , any update on this?
The issue is still happening:
![image](https://user-images.githubusercontent.com/5447190/236912675-22a29067-c644-42cb-81df-54d54dc80383.png)
